plugins/scale/scale.cpp

```diff
@@ -787,6 +787,43 @@ class wayfire_scale : public wf::plugin_interface_t
             target_alpha);
     }
 
+    static bool view_compareX(const wayfire_view& A, const wayfire_view& B)
+    {
+        auto vgA = A->get_wm_geometry();
+        auto vgB = B->get_wm_geometry();
+        if (vgA.x == vgB.x)
+        {
+            return vgA.y < vgB.y;
+        }
+
+        return vgA.x < vgB.x;
+    }
+
+    static bool view_compareY(const wayfire_view& A, const wayfire_view& B)
+    {
+        auto vgA = A->get_wm_geometry();
+        auto vgB = B->get_wm_geometry();
+        if (vgA.y == vgB.y)
+        {
+            return vgA.x < vgB.x;
+        }
+
+        return vgA.y < vgB.y;
+    }
+
+    static void view_sort(std::vector<wayfire_view>& views, int wcnt)
+    {
+        std::sort(views.begin(), views.end(), view_compareY);
+        int n = (int)views.size();
+        int i = 0, j;
```

Comment:
No need for casts if we use native types.

```suggestion
        size_t i = 0, j;
```

Reply:
done